### PR TITLE
Use UI to create providers

### LINKED ISSUE #27: Automate creating Providers using UI
With new support added in https://github.com/rancher/capi-ui-extension/issues/13, use UI to create provider instead of importing YAML

### PR DESCRIPTION
**Pull request by cpinjani**
### What does this PR do?
Automate Creating CAPI providers by UI currently being used by tests

### Which issue(s) this PR fixes (optional, in fixes #<issue number>(, fixes #<issue_number>, ...) format, will close the issue(s) when PR gets merged):
Fixes #27 

### Checklist:
- [x] GitHub Actions: https://github.com/rancher-sandbox/rancher-turtles-e2e/actions/runs/8801252565

### CODE REVIEW

File: tests/cypress/latest/support/commands.ts [hunk 1]
```diff
@@ -34,12 +34,8 @@ Cypress.Commands.add('namespaceAutoImport', (mode) => {
 
   // Select default namespace
   cy.setNamespace('Project: Default');
-
-  // Reload required since kebab menu icon not clickable
-  cy.reload(true);
```
Comment (cpinjani) [inline tests/cypress/latest/support/commands.ts:39]: It was issue with v2.7, so not applicable now

File: .github/workflows/master-e2e.yaml [hunk 2]
```diff
@@ -17,8 +17,11 @@ on:
         # - if the person still has access to the repo
         description: PAT token used to add runner
         required: true
-      aws_b64encoded_credentials:
```
Comment (cpinjani) [inline .github/workflows/master-e2e.yaml:20]: AWS Cloud credentials mapping with Turtles handles secret configuration

File: tests/cypress/latest/support/commands.ts [hunk 3]
```diff
@@ -92,10 +88,72 @@ Cypress.Commands.add('checkCAPICluster', (clusterName) => {
...
+  cy.accesMenuSelection('Cluster Management', 'CAPI');
+  cy.contains('Infrastructure Providers').click();
+  cy.clickButton('Create');
+  cy.contains(type, { matchCase: false }).click();
+  cy.contains('Infrastructure Provider: Create ' + type, { matchCase: false }).should('be.visible');
+
+  // TODO: Add varibales support after capi-ui-extension/issues/49
+  cy.get('.vs__selected-options').eq(0)
+    .type(namespace + '{enter}');
+  cy.get('.vs__selected-options > span.vs__selected').eq(0).should('contain', namespace);
```
Comment (valaparthvi) [inline tests/cypress/latest/support/commands.ts:129]:
```suggestion
    cy.getBySel('name-ns-description-namespace').type(namespace + '{enter}');
```

File: tests/cypress/latest/support/commands.ts [hunk 4]
```diff
@@ -92,10 +88,72 @@ Cypress.Commands.add('checkCAPICluster', (clusterName) => {
...
+  cy.typeValue('Name', name);
+
+  // Select Cloud credntials name
+  if (type != 'docker') {
+    cy.get('.vs__selected-options > span.vs__selected').eq(1).should('contain', cloudCredentials);
```
Comment (valaparthvi) [inline tests/cypress/latest/support/commands.ts:134]:
```suggestion
    cy.getBySel('cluster-prov-select-credential').trigger('click');
    cy.contains(cloudCredentials).click();
```

File: tests/cypress/latest/support/commands.ts [hunk 5]
```diff
@@ -92,10 +88,72 @@ Cypress.Commands.add('checkCAPICluster', (clusterName) => {
...
+
+  // Select provider type
+  cy.get('.vs__selected-options').eq(1)
+    .click();
+  cy.contains(type).click();
+  cy.get('.vs__selected-options > span.vs__selected').eq(1).should('contain', type);
+
+  cy.get('.vs__selected-options').eq(0)
+    .type(namespace + '{enter}');
+  cy.get('.vs__selected-options > span.vs__selected').eq(0).should('contain', namespace);
```
Comment (valaparthvi) [inline tests/cypress/latest/support/commands.ts:107]:
```suggestion
      cy.getBySel('name-ns-description-namespace').type(namespace + '{enter}');
```

File: tests/cypress/latest/support/commands.ts [hunk 6]
```diff
@@ -92,10 +88,72 @@ Cypress.Commands.add('checkCAPICluster', (clusterName) => {
...
+  cy.accesMenuSelection('Cluster Management', 'CAPI');
+  cy.contains('Infrastructure Providers').click();
+  cy.clickButton('Create');
+  cy.contains('Custom').click();
+
+  // Select provider type
+  cy.get('.vs__selected-options').eq(1)
+    .click();
+  cy.contains(type).click();
+  cy.get('.vs__selected-options > span.vs__selected').eq(1).should('contain', type);
```
Comment (valaparthvi) [inline tests/cypress/latest/support/commands.ts:103]:
```suggestion
  cy.contains("Provider type").click();
  cy.contains(type).click();
```

**Review by valaparthvi — CHANGES_REQUESTED** [review]

**Comment by cpinjani** [comment]
CI passed with latest changes: https://github.com/rancher-sandbox/rancher-turtles-e2e/actions/runs/8817437865

**Pull request merged**